src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1481,6 +1490,42 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
 
         return null;
     }
+    
+    /**
+     * Add log event for request failure with a cause. 
+     */
+    private void addLogEventForRequestFailureWithCause(final String eventName, final AuthenticationRequest request, 
+            final AuthenticationException authenticationException) {
+        ClientAnalytics.logEvent(
+                eventName,
+                new InstrumentationPropertiesBuilder(request, authenticationException).build());
+    }
+    
+    /**
+     * @return True if access token is returned from result, false otherwise. 
+     */
+    private boolean isAccessTokenReturned(final AuthenticationResult authResult) {
+        return authResult != null && !StringExtensions.IsNullOrBlank(authResult.getAccessToken());
+    }
+    
+    /**
+     * If access token redeem with family refresh token fails, should fall back to MRRT if existed. 
+     */
+    private AuthenticationResult fallbackToMRRT(final AuthenticationRequest authRequest, final RefreshItem refreshItem) 
```

Comment:
Because it looks weird when you have getRT and get AT in one way, and then fallback scenario getRT and getAt in another method.